### DISCUSSION TITLE
PIP-7: Checking Timestamp Difference in Handshaking

### DISCUSSION THREAD

**Discussion by ja7ad** [pactus-project/PIPs#22] [PIPs]
PIP-7: [Checking Timestamp Difference in Handshaking](https://pips.pactus.org/PIPs/pip-7)

**Comment by b00f** [comment]
@kehiy 

Based on this proposal, if the `|now() - hello_message.my_time| > 20`, the `hello_message` is rejected. By rejecting the `hello_message`, two nodes cannot communicate anymore.
20 seconds is the time of 2 blocks, in the worse case scenario, one of the node may start querying about one missed blocks.
They sync module can [tolerate](https://github.com/pactus-project/pactus/blob/a8d5a56bf28a1b36408c53a15c9cad6fe187e64f/sync/sync.go#L249) one block, not two. 
I suggest reduce it to 10 seconds, and heretofore, in worse case scenario, the nodes won't start querying the latest blocks.

**Reply by kehiy** [reply]
that's make sense, I will reduce it to 10 seconds.
